src/definitions/bluebps.json

```diff
@@ -0,0 +1,276 @@
+                "id": "9001",
+                "description": "Set the under-voltage timeout",
+                "payload": [
+                    {
+                        "name": "timeout",
+                        "type": "u16",
+                        "description": "If an individual cell exceeds the configured limit for this duration of time, the power will be locked-out",
+                        "units": "millisecond"
+                    }
+                ]
+            },
+            "current_max": {
+                "id": "9002",
+                "description": "Set the maximum allowed battery current",
+                "payload": [
+                    {
+                        "name": "limit",
+                        "type": "u16",
+                        "description": "The maximum allowed battery current 0~20000 = 0~200A",
+                        "units": "centiampere"
+                    }
+                ]
+            },
+            "current_timeout": {
+                "id": "9003",
+                "description": "Set the over-current timeout",
+                "payload": [
+                    {
+                        "name": "timeout",
+                        "type": "u16",
+                        "description": "If the battery current exceeds the configured limit for this duration of time, the power will be locked-out",
+                        "units": "millisecond"
+                    }
+                ]
+            },
+            "temperature_max": {
+                "id": "9004",
+                "description": "Set the maximum allowed battery temperature",
+                "payload": [
+                    {
+                        "name": "limit",
+                        "type": "u16",
+                        "description": "The maximum temperature allowed at the thermistor probe installed on the battery. 0~5000: 0~5V",
+                        "units": "centidegrees Centigrade"
+                    }
+                ]
+            },
+            "temperature_timeout": {
+                "id": "9005",
+                "description": "Set the over-temperature timeout",
+                "payload": [
+                    {
+                        "name": "timeout",
+                        "type": "u16",
+                        "description": "If the battery temperature exceeds the configured limit for this duration of time, the power will be locked-out",
+                        "units": "millisecond"
+                    }
+                ]
+            },
+            "stream_rate": {
+                "id": "9006",
+                "description": "Set the frequency to automatically output state messages.",
+                "payload": [
+                    {
+                        "name": "rate",
+                        "type": "u32",
+                        "description": "Rate to stream `state` messages. 0~100000Hz",
+                        "units": "Hz"
+                    }
```

Comment:
Can we specify the message or it's only possible to get "state" (9106) message ?

Reply:
Right now is state only. This is the only place I see the use case.

Many parts of this protocol have not been implemented in firmware yet.

Reply:
we can add other messages to stream if we find the need.